src/Protocol/ForgeMods.cpp

```diff
@@ -0,0 +1,89 @@
+
+// ForgeMods.cpp
+
+// Data structure listing the Forge mods name and versions
+
+#include "Globals.h"
+#include "ForgeMods.h"
+
+cForgeMods::cForgeMods()
+{
+}
+
+
+
+
+
+cForgeMods::cForgeMods(AStringMap a):m_Mods(a)
+{
+	for (const auto & item: a)
+	{
+		m_ModNames.push_back(item.first);
+	}
+}
```

Comment:
I think the issue (correct me if I'm wrong) is actually in
```
cForgeMods::cForgeMods(AStringMap a):m_Mods(a)
```
When the cForgeMods constructor is called, like so:
```
cForgeMods ModList(MyMapOfModThings);
```
The variable MyMapOfModThings is copied into the stack space of the constructor method, where it is copied again to the m_Mods member variable (via the map constructor).

If you do:
```
cForgeMods::cForgeMods(const AStringMap & a):m_Mods(a)
```
Then you skip the first copy, and it copies directly from MyMapOfModThings into the member variable.

(actually, I think just passing by reference does that as well, but you don't modify a, so might as well pass it as const)